shared/modules/typed-signature.test.ts

```diff
@@ -0,0 +1,113 @@
+import { MessageTypes } from '@metamask/eth-sig-util';
+import { sanitizeMessageRecursively } from './typed-signature';
+
+describe('typed-signature utils', () => {
+  describe('sanitizeMessageRecursively', () => {
+    const types: MessageTypes = {
+      EIP712Domain: [
+        { name: 'name', type: 'string' },
+        { name: 'version', type: 'string' },
+        { name: 'chainId', type: 'uint256' },
+        { name: 'verifyingContract', type: 'address' },
+      ],
+      Root: [
+        { name: 'root', type: 'bytes32' },
+        { name: 'nested', type: 'NestedType' },
+      ],
+      NestedType: [{ name: 'field', type: 'string' }],
+    };
+
+    const primaryType = 'Root';
```

Comment:
Minor, capitals and underscore for constants?

Reply:
Done https://github.com/MetaMask/metamask-extension/pull/29698/commits/cdb8f6de50e619e0424689b4855cc0cfed60c6b0